src/solver/coeffs.jl

```diff
@@ -0,0 +1,327 @@
+#  TODO(Alex): it is not clear to me at the moment if we should use
+# term or TermInterface.maketerm
+
+"""
+    sub(subs, place_to_sub)
+Helper function for filter_poly. Generates a symbolics variable and adds it
+to subs (changes the original array passed to the sub function). returns the
+subbed value so the filter_poly function could change it in its scope.
+
+# Arguments
+- subs: Vector of dicts which consist of var_subbed => value_subbed in filter_poly
+- place_to_sub: The place which should be substituted by a new variable.
+
+"""
+function sub(subs, place_to_sub)
+    sub_var = gensym()
+    sub_var = (@variables $sub_var)[1]
+
+    subs[sub_var] = place_to_sub
+    place_to_sub = sub_var.val
+
+    return place_to_sub
+end
+
+"""
+    clean_f(filtered_expr, var, subs)
+
+Helper function for `filter_poly` which is called directly before returning
+the `filtered_expression`. This function aims to get the filtered expressions
+resulting from `filter_poly` ready to get used by solve, `get_roots`, and any other
+function in the library. An important feature of it is that it simplifies fractions
+to make the output "valid" in the eyes of `factor_use_nemo` and other Nemo functions.
+
+# Arguments
+- `filtered_expr`: The output from `_filter_poly`.
+- var: The variable which is filtered for.
+- subs: Vector of dicts which consist of `var_subbed` => `value_subbed` in `_filter_poly`.
+
+"""
+function clean_f(filtered_expr, var, subs)
+    filtered_expr = simplify_fractions(simplify(real(filtered_expr)))
+    unwrapped_f = unwrap(filtered_expr)
+    !iscall(unwrapped_f) && return filtered_expr
+    oper = operation(unwrapped_f)
+
+    if oper === (/)
+        args = arguments(unwrapped_f)
+        if any(isequal(var, x) for x in get_variables(args[2]))
+            return filtered_expr
+        end
+        filtered_expr = args[1]
+        @info substitute(args[2], subs, fold=false) != 0
+    end
+    return filtered_expr
+end
+
+"""
+    filter_stuff(expr)
+Helper function for _filter_poly and filter_poly which aims to filter
+constants and things that dont contain variables like term(sqrt, 2).
+Filters only if the passed expr is scary (i.e. not a Rational or Int).
+
+# Arguments
+- expr: The detected constant in _filter_poly
+
+# Examples
+```jldoctest
+julia> filter_stuff(term(sqrt, 2))
+(Dict{Any, Any}(var"##278" => sqrt(2)), var"##278")
+
+julia> filter_stuff(123)
+(Dict{Any, Any}(), 123)
+```
+"""
+function filter_stuff(expr)
+
+    if expr isa Integer || expr isa Rational
+        return Dict(), expr
+
+    else
+        expr = isequal(expr, true) ? 1 : expr
```

Comment:
why replace `true` with `1` ?

Reply:
i think it caused some weird behavior during computation?